src/workers/newView.ts

```diff
@@ -19,60 +22,24 @@ const addView = async (con: DataSource, entity: View): Promise<boolean> => {
     !existing ||
     entity.timestamp.getTime() - existing.timestamp.getTime() > ONE_WEEK
   ) {
-    await repo.save(entity);
-    return true;
+    const viewRecord = await repo.save(entity);
+    return { view: viewRecord, isNewView: true };
   }
-  return false;
+  return { view: existing, isNewView: false };
```

Comment:
I think if we extract the `isNewView` into the upper scope, the code would look nicer. E.g.:
```typescript
let view = existing;
const isNewView = !existing ||
    entity.timestamp.getTime() - existing.timestamp.getTime() > ONE_WEEK
if (isNewView) {
    view = await repo.save(entity)
}
return { view, isNewView };
```

Reply:
sure thing, good point 👍